packages/visualizations/src/components/Chart/Chart.svelte

```diff
@@ -52,13 +164,51 @@
 
     function chartJsDataLabels(dataLabels: DataLabelsConfiguration | undefined): DataLabelsOptions {
         if (dataLabels === undefined) return { display: false };
+        const formatter = dataLabels.formatter;
+        const aligner = dataLabels.align;
+        const anchorer = dataLabels.anchor;
         return {
-            align: defaultValue(dataLabels.align, 'center'),
+            align: aligner
+                ? (context) => {
+                      return aligner(context.dataIndex, { dataFrame });
+                  }
+                : 'end',
+            anchor: anchorer
+                ? (context) => {
+                      return anchorer(context.dataIndex, { dataFrame });
+                  }
+                : 'end',
             display: defaultValue(dataLabels.display, false),
-            color: chartJsColorPalette(dataLabels.color),
-            backgroundColor: chartJsColorPalette(dataLabels.backgroundColor),
-            offset: defaultValue(dataLabels.offset, 0),
-            borderRadius: defaultValue(dataLabels.borderRadius, 0),
+            color: (context) => {
+                if (context.dataset.borderColor) {
+                    return context.dataset.borderColor;
+                } else if (dataLabels.color) {
+                    return chartJsColorPalette(dataLabels.color);
+                } else {
+                    return 'rgb(0, 0, 0)';
+                }
+            },
+            backgroundColor: (context) => {
+                if (options.series[0]?.type === 'pie') {
+                    return context.dataset.backgroundColor[context.dataIndex];
+                } else {
+                    defaultValue(
+                        chartJsColorPalette(dataLabels.backgroundColor),
+                        'rgb(255,255,255)'
+                    );
+                }
+            },
+            offset: defaultValue(dataLabels.offset, 4),
+            borderRadius: defaultValue(dataLabels.borderRadius, 3),
+            formatter: formatter
+                ? (value, context) => {
+                      const formattedDataFrame = handleLongDataLabels(dataFrame);
+                      return formatter(context.dataIndex, formattedDataFrame);
```

Comment:
I probably missed something but I'm a little bit confuse regarding `dataFrame`:

In `DataLabelsConfiguration` the `formatter` is defined as such:
```typescript
formatter?: (index:number, context:{ dataFrame:DataFrame }) => string | string[];
```

according to this, shouldn't we pass the argument this way instead?

```typescript
return formatter(context.dataIndex, { dataFrame: formattedDataFrame });
```

But after a few tests, it seems that the type declaration should change into:

```typescript
formatter?: (index:number, dataFrame:DataFrame) => string | string[];
```